File: microservices/feedback-service/src/main/java/com/karate/feedback_service/infrastructure/client/TrainingSessionClient.java
package com.karate.feedback_service.infrastructure.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "training-service")
public interface TrainingSessionClient {
    @GetMapping("/training-sessions/{sessionId}/exists")
    Boolean checkTrainingSessionExists(@PathVariable Long sessionId);
}

File: microservices/feedback-service/src/main/kotlin/com/karate/feedback_service/infrastructure/client/TrainingSessionClient.kt
package com.karate.feedback_service.infrastructure.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "training-service")
interface TrainingSessionClient {
    @GetMapping("/internal/trainings/{trainingId}/exists")
    fun checkTrainingExists(@PathVariable trainingId: Long): Boolean?
}
